### DISCUSSION TITLE
Shanghai hardfork not supported, invalid opcodes in solidity 0.8.20+ generated code

### DISCUSSION THREAD

**Discussion by reachjason** [celo-org/celo-blockchain#2268] [Feature requests (Needs)]
## Expected Behavior

The smart contract deploys successfully

## Actual Behavior

 i get the following error : "invalid opcode: opcode 0x5f not defined"

## Steps to reproduce the behavior

Try deploying a contract using remix and latest solidity versions.

**Comment by reachjason** [comment]
More like a feature request. Reference:

https://ethereum.stackexchange.com/questions/150281/invalid-opcode-opcode-0x5f-not-defined

**Comment by palango** [comment]
Hi @reachjason,
thanks for reporting this. This is due to [EIP-3855](https://eips.ethereum.org/EIPS/eip-3855) not being enabled on Celo yet. 

For now please resort to setting the EVM version explicitly: https://docs.soliditylang.org/en/v0.8.20/using-the-compiler.html#setting-the-evm-version-to-target
